src/mon/MDSMonitor.cc

```diff
@@ -860,6 +839,32 @@ bool MDSMonitor::prepare_beacon(MonOpRequestRef op)
     }
   }));
 
+  return true;
+
+evict:
+  if (!mon.osdmon()->is_writeable()) {
+    dout(1) << __func__ << ": waiting for writeable OSDMap to evict" << dendl;
+    mon.osdmon()->wait_for_writeable(op, new C_RetryMessage(this, op));
+    return false;
+  }
+
+  fail_mds_gid(pending, gid);
+  request_proposal(mon.osdmon());
+  dout(5) << __func__ << ": pending map now:" << dendl;
+  print_map(pending);
+
+  goto null;
```

Comment:
This falls-through to "null" block.

BTW, is there a need to reply back after eviction?

Reply:
It's just defensive coding (for future go-tos) and making the fall-through clear.

The null map will cause the MDS to respawn and rejoin as standby immediately. Otherwise, the MDS will just keep the connection open with the monitors and just report laggy beacons.

Reply:
> t's just defensive coding (for future go-tos) and making the fall-through clear.

I would just leave a comment there.

> 
> The null map will cause the MDS to respawn and rejoin as standby immediately. Otherwise, the MDS will just keep the connection open with the monitors and just report laggy beacons.

OK, fair enough.